mbuild/tests/test_utils.py

```diff
@@ -186,3 +187,325 @@ def test_has_ipython(self):
 
     def test_removed_func_error(self):
         RemovedFuncError("old_function", "new_function", "0.0.0", "0.1.0")
+
+
+class TestUtilsConversion(BaseTest):
+    def test_RB_to_OPLS_c5_not_0(self):
+        with pytest.raises(
+            ValueError,
+            match=r"ERROR: c5 must equal zero, so this conversion is not possible.",
+        ):
+            c0 = 0.1
+            c1 = 0.1
+            c2 = -0.2
+            c3 = -0.1
+            c4 = -0.2
+            c5 = 0.3
+            RB_to_OPLS(c0, c1, c2, c3, c4, c5)
+
+    def test_RB_to_OPLS_f0_not_0(self):
+        with pytest.raises(
+            ValueError,
+            match=r"ERROR: f0 \= 2 \* \(c0 \+ c1 \+ c2 \+ c3 \+ c4 \+ c5\) must equal zero, "
+            "so this conversion is not possible.",
+        ):
+            c0 = 0.4
+            c1 = 0.4
+            c2 = -0.1
+            c3 = 0.4
+            c4 = -0.2
+            c5 = 0
+            RB_to_OPLS(c0, c1, c2, c3, c4, c5)
+
+    def test_RB_to_OPLS_error_tol_not_float(self):
+        with pytest.raises(
+            TypeError,
+            match=r"ERROR: The error_tol variable must be a float.",
+        ):
+            c0 = 0.1
+            c1 = 0.1
+            c2 = -0.2
+            c3 = -0.1
+            c4 = -0.2
+            c5 = 0.3
+            RB_to_OPLS(c0, c1, c2, c3, c4, c5, error_tol="s")
+
+    def test_RB_to_OPLS_and_back_random_values(self):
```

Comment:
Lets convert this to a parametrized pytest test, so we can add additional tests if needed without replicating a lot of code.

Note, the spacing is a bit off in my sample code below, but the setup should be correct.

```python3
    @pytest.mark.parametrize(
                             "c0, c1, c2, c3, c4, c5",
                                 [
                                    (-8.2723, 0.2263, 10.22, -3.208, 1.034, 0.0),
                                     # put the rest of the tests here, each as a tuple
                                  ],
    )
    def test_RB_to_OPLS_and_back_random_values(c1, c2, c3, c4, c5):
        # Note the sum of c0 to c5 must be zero for these test.
        # However, this may not be true for real dihedrals.

        # Test Number 1
        c0 = -8.2723
        c1 = 0.2263
        c2 = 10.22
        c3 = -3.208
        c4 = -(c0 + c1 + c2 + c3)
        c5 = 0

        opls_coeffs = RB_to_OPLS(c0, c1, c2, c3, c4, c5)
        reversed_RB_coeffs = OPLS_to_RB(
            opls_coeffs[0],
            opls_coeffs[1],
            opls_coeffs[2],
            opls_coeffs[3],
        )

        assert bool(np.isclose(c0, reversed_RB_coeffs[0], atol=1e-10, rtol=0))
        assert bool(np.isclose(c1, reversed_RB_coeffs[1], atol=1e-10, rtol=0))
        assert bool(np.isclose(c2, reversed_RB_coeffs[2], atol=1e-10, rtol=0))
        assert bool(np.isclose(c3, reversed_RB_coeffs[3], atol=1e-10, rtol=0))
        assert bool(np.isclose(c4, reversed_RB_coeffs[4], atol=1e-10, rtol=0))
        assert bool(np.isclose(c5, reversed_RB_coeffs[5], atol=1e-10, rtol=0))

```

Reply:
[An example here in test_lattice.py](https://github.com/mosdef-hub/mbuild/blob/fa53246c16c4326c759c9a9a091f1f1f9c77840d/mbuild/tests/test_lattice.py#L59-L71)

Reply:
In the example above. You don’t have to define c0-5 in the code body, that was a mistake of me copying source code over. Those variables are defined in the function args and by the parameterize decorator

Reply:
Changed